### DISCUSSION TITLE
Teknik Imputasi - Apabila ada suatu kolom dari data kategorikal missing sebanyak 80%-90%, cara imputasi seperti apa?

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#34] [Q&A] [answered]
Jika ada data kategorikal missing sebanyak 80%-90% dalam lingkup suatu kolom dari beberapa kolom dalam suatu data frame, apakah sebaiknya dibuang saja kolom itu, atau bisa diimputasi dengan membuat kategori baru dengan cara clustering dari data di sekitarnya?

Jika kemudian setelah data itu utuh akan dijadikan sebagai train set untuk suatu klasifikasi kolom lain, apakah imputasi dengan clustering tersebut akan berpotensi menjadi masalah?

**Comment by dwiknrd** [comment, marked as answer]
Halo [Hertzspr](https://github.com/Hertzspr) treatment missing value memang berbeda sesuai dengan kasus pada data yang kita punya. Beberapa treatment missing value pada data kategorikal data adalah sebagai berikut:

1. Apabila observasi pada kolom tersebut dapat dicari di sumber lain atau database maka nilai missing dapat terisi,
2. Apabila kolom tersebut tidak berpengaruh terhadap analisis maka kita dapat menghapus observasi yang memiliki nilai missing, karena nilai missing akan menyebabkan masalah pada step selanjutnya,
3. Apabila kolom tersebut sangat dibutuhkan untuk analisis anda dapat melakukan imputasi dengan membuat kategori baru, namun hal ini bisa dilakukan apabila anda tahu category apa saja yang ada pada kolom tersebut. Pada kasus ini karena anda menggunakan machine learning sebelum menjadikan data untuk model machine learning selanjutnya tentu data yang digunakan bukanlah data yang sebenarnya, sehingga apabila ada data baru yang perlu diprediksi akan menghasilkan hasil yang kurang baik.

Semoga membantu 😀

**Reply by Hertzspr** [reply]
Untuk spesifiknya, sebenernya lagi ngeliatin kolom Cabin di data Titanic yang di Kaggle, itu banyak banget bolongnya, bingung, saya hapus aja dah kayaknya.

**Reply by dwiknrd** [reply]
Halo [Hertzspr](https://github.com/Hertzspr) ketika sebuah kolom memiliki nilai-nilai yang missing sangat banyak, maka dapat dikatakan bahwa tidak ada gunanya bagi kita untuk melakukan imputasi. 
Jika pada kolom cabin memiliki nilai missing lebih dari 80%, anda dapat menghapus kolom tersebut dari analisis anda. Pastikan bahwa kolom tersebut tidak penting atau tidak dibutuhkan untuk analisis. 

Berikut saya lampirkan artikel analisis data titanic beserta treatment missing value https://www.analyticsvidhya.com/blog/2021/10/end-to-end-introduction-to-handling-missing-values/
